ccl/changefeedccl: TestChangefeedJobUpdateFailsIfNotClaimed failed

ccl/changefeedccl.TestChangefeedJobUpdateFailsIfNotClaimed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19506672?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19506672?buildTab=artifacts#/) on master @ [18b42de2ab0bd242c0fa9334d1ba1e90566d7683](https://github.com/cockroachdb/cockroach/commits/18b42de2ab0bd242c0fa9334d1ba1e90566d7683):


```
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/8db35d09d3aa97ee0974c3c6b3b5529d/logTestChangefeedJobUpdateFailsIfNotClaimed326187133
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:6195: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db35d09d3aa97ee0974c3c6b3b5529d/logTestChangefeedJobUpdateFailsIfNotClaimed326187133
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed (5.74s)
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed/kafka
    helpers_test.go:1043: making server as system tenant
    helpers_test.go:1119: making kafka feed factory
    changefeed_test.go:6189: expected distflow to fail
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed/kafka (5.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedJobUpdateFailsIfNotClaimed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-49604

ccl/changefeedccl.TestChangefeedJobUpdateFailsIfNotClaimed [failed](https://tanzanite.cluster.engflow.com/invocations/default/1ef03b3e-6707-4d6c-a25d-70bbff6163fe?testReportRun=2&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [70bb4fafea5534d34fab9440975da5526933c955](https://github.com/cockroachdb/cockroach/commits/70bb4fafea5534d34fab9440975da5526933c955):


```
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedJobUpdateFailsIfNotClaimed903043594
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:6195: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedJobUpdateFailsIfNotClaimed903043594
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed (5.66s)
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed/kafka
    helpers_test.go:1043: making server as system tenant
    helpers_test.go:1119: making kafka feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:6189: expected distflow to fail
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed/kafka (5.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedJobUpdateFailsIfNotClaimed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedJobUpdateFailsIfNotClaimed [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19510583?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19510583?buildTab=artifacts#/) on master @ [acfdcc8e2e5da1d4d37fee05f41a067e855de912](https://github.com/cockroachdb/cockroach/commits/acfdcc8e2e5da1d4d37fee05f41a067e855de912):


```
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/8db35d09d3aa97ee0974c3c6b3b5529d/logTestChangefeedJobUpdateFailsIfNotClaimed1212490235
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:6195: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db35d09d3aa97ee0974c3c6b3b5529d/logTestChangefeedJobUpdateFailsIfNotClaimed1212490235
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed (6.31s)
=== RUN   TestChangefeedJobUpdateFailsIfNotClaimed/cloudstorage
    helpers_test.go:1043: making server as system tenant
    helpers_test.go:1119: making cloudstorage feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:6189: expected distflow to fail
--- FAIL: TestChangefeedJobUpdateFailsIfNotClaimed/cloudstorage (6.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedJobUpdateFailsIfNotClaimed.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -6132,8 +6132,6 @@ func TestChangefeedJobUpdateFailsIfNotClaimed(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 144912)
-
 	// Set TestingKnobs to return a known session for easier
 	// comparison.
 	adoptionInterval := 20 * time.Minute
@@ -6175,9 +6173,11 @@ func TestChangefeedJobUpdateFailsIfNotClaimed(t *testing.T) {
 		// another node.
 		sqlDB.Exec(t, `UPDATE system.jobs SET claim_session_id = NULL WHERE id = $1`, jobID)
 
-		timeout := 5 * time.Second
+		timeout := (5 * time.Second) + changefeedbase.Quantize.Get(&s.Server.ClusterSettings().SV)
+
 		if util.RaceEnabled {
-			timeout = 30 * time.Second
+			// Timeout should be at least 30s to allow for race conditions.
+			timeout += 25 * time.Second
 		}
 		// Expect that the distflow fails since it can't
 		// update the checkpoint.

